cmd/find.go

```diff
@@ -0,0 +1,454 @@
+		str = strings.Replace(str, "{time}", t, -1)
+	}
+
+	// replace all instances of {""}
+	if strings.Contains(str, "{\""+"\"}") {
+		str = strings.Replace(str, "{\""+"\"}", strconv.Quote(path), -1)
+	}
+
+	// replace all instances of {"base"}
+	if strings.Contains(str, "{\""+"base"+"\"}") {
+		str = strings.Replace(str, "{\""+"base"+"\"}", strconv.Quote(filepath.Base(path)), -1)
+	}
+
+	// replace all instances of {"dir"}
+	if strings.Contains(str, "{\""+"dir"+"\"}") {
+		str = strings.Replace(str, "{\""+"dir"+"\"}", strconv.Quote(filepath.Dir(path)), -1)
+	}
+
+	if strings.Contains(str, "{\""+"url"+"\"}") {
+		s := GetPurl(path)
+		str = strings.Replace(str, "{\""+"url"+"\"}", strconv.Quote(s), -1)
+	}
+
+	if strings.Contains(str, "{\""+"size"+"\"}") {
+		s := humanize.IBytes(uint64(fileContent.Size))
+		str = strings.Replace(str, "{\""+"size"+"\"}", strconv.Quote(s), -1)
+	}
+
+	if strings.Contains(str, "{\""+"time"+"\"}") {
+		str = strings.Replace(str, "{\""+"time"+"\"}", strconv.Quote(fileContent.Time.String()), -1)
+	}
+
+	return str
+}
+
+// WatchHelper used in conjunction with WatchFlag method to detect and preform desired action when an object is created
+func WatchHelper(clnt Client, params watchParams, ctx *cli.Context, alias string, fileContent contentMessage) {
+	watchObj, err := clnt.Watch(params)
+	fatalIf(err, "Cannot watch with given params")
+
+	// get client url and remove
+	cliTemp, _ := getHostConfig(alias)
+	cliTempURL := cliTemp.URL
+
+	// enables users to kill using the control + c
+	trapCh := signalTrap(os.Interrupt, syscall.SIGTERM)
+
+	wg := sync.WaitGroup{}
+	wg.Add(1)
+
+	// opens a channel of all content created on the server, any errors detected,
+	// and scanning for the user input (Control + C)
+	go func() {
+		defer wg.Done()
+
+		for {
+			select {
+			case <-trapCh:
+				console.Println()
+				close(watchObj.doneChan)
+				return
+			case event, ok := <-watchObj.Events():
+				if !ok {
+					return
+				}
+				msg := contentMessage{}
+				msg.Key = strings.TrimPrefix(event.Path, cliTempURL)
+				msg.Time, _ = time.Parse(time.RFC822, event.Time)
+				msg.Size = event.Size
+
+				msg.Key = alias + msg.Key
+
+				// check to see if the newly create object matches the given params
+				match := isMatch(msg, ctx.Bool("or"), ctx)
+
+				if match && ctx.String("exec") != "" {
+					ExecFlag(ctx, msg.Key, fileContent)
+				}
+
+			case err, ok := <-watchObj.Errors():
+				if !ok {
+					return
+				}
+				errorIf(err, "Unable to watch for events.")
+				return
+			}
+		}
+	}()
+
+	wg.Wait()
+}
+
+func isMatch(fileContent contentMessage, orOp bool, ctx *cli.Context) bool {
+	match := true
+
+	if (match && !orOp) && ctx.String("ignore") != "" {
+		match = !PathFlag(fileContent.Key, ctx.String("ignore"))
+	}
+
+	// verify that the newly added object matches all of the other specified params
+	if ctx.String("name") != "" {
+		temp1, err1 := NameFlag(fileContent.Key, ctx.String("name"))
+		match = temp1
```

Comment:
Don't you think lets say name and ignore were provided if `match` is set to true by PathFlag() before would be overwritten by the NameFlag which can lead to further issues down the line.